--- robots.txt	2019-09-02 11:19:08.000000000 +0530
+++ robots_new	2019-09-02 11:26:17.000000000 +0530
@@ -14,25 +14,26 @@
 # http://www.robotstxt.org/robotstxt.html
 
 User-agent: *
+Disallow: /
 # CSS, JS, Images
-Allow: /core/*.css$
-Allow: /core/*.css?
-Allow: /core/*.js$
-Allow: /core/*.js?
-Allow: /core/*.gif
-Allow: /core/*.jpg
-Allow: /core/*.jpeg
-Allow: /core/*.png
-Allow: /core/*.svg
-Allow: /profiles/*.css$
-Allow: /profiles/*.css?
-Allow: /profiles/*.js$
-Allow: /profiles/*.js?
-Allow: /profiles/*.gif
-Allow: /profiles/*.jpg
-Allow: /profiles/*.jpeg
-Allow: /profiles/*.png
-Allow: /profiles/*.svg
+#Allow: /core/*.css$
+#Allow: /core/*.css?
+#Allow: /core/*.js$
+#Allow: /core/*.js?
+#Allow: /core/*.gif
+#Allow: /core/*.jpg
+#Allow: /core/*.jpeg
+#Allow: /core/*.png
+#Allow: /core/*.svg
+#Allow: /profiles/*.css$
+#Allow: /profiles/*.css?
+#Allow: /profiles/*.js$
+#Allow: /profiles/*.js?
+#Allow: /profiles/*.gif
+#Allow: /profiles/*.jpg
+#Allow: /profiles/*.jpeg
+#Allow: /profiles/*.png
+#Allow: /profiles/*.svg
 # Directories
 Disallow: /core/
 Disallow: /profiles/
